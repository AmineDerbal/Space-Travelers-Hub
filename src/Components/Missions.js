import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/Missions';
import './styles/Missions.css';
import Mission from './Mission';
import Loader from './Loader';
import 'bootstrap/dist/css/bootstrap.css';

const Missions = () => {
  const dispatch = useDispatch();
  const { missions, isLoading } = useSelector((store) => store.missions);
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missions.length]);
  if (isLoading) {
    return <Loader />;
  }
  return (
    <div className="container-fluid px-5">
      <table className="table table-hover table-striped table-responsive">
        <thead>
          <tr>
            <th className="heading" scope="col">Missions</th>
            <th className="heading" scope="col">Description</th>
            <th className="heading" scope="col">Status</th>
            <th className="heading" scope="col">{' '}</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <Mission
              name={mission.name}
              key={mission.id}
              id={mission.id}
              description={mission.description}
              reserved={mission.reserved}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
