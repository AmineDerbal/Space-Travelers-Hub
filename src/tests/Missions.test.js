import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../Components/Missions';

it('renders Missions correctly', () => {
  const tree = renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <Router>
          <Missions />
        </Router>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
