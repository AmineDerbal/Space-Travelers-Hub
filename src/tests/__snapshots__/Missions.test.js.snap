// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Missions correctly 1`] = `
<div
  className="container-fluid px-5"
>
  <table
    className="table table-hover table-striped table-responsive"
  >
    <thead>
      <tr>
        <th
          className="heading"
          scope="col"
        >
          Missions
        </th>
        <th
          className="heading"
          scope="col"
        >
          Description
        </th>
        <th
          className="heading"
          scope="col"
        >
          Status
        </th>
        <th
          className="heading"
          scope="col"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
